model/example_domains_test.go

```diff
@@ -0,0 +1,196 @@
+package model_test
+
+import (
+	"fmt"
+
+	"github.com/nextmv-io/sdk/model"
+)
+
+func ExampleDomains_add() {
+	d1 := model.Repeat(3, model.Singleton(42))
+	d2 := d1.Add(1, 41, 43)
+	fmt.Println(d2)
+	// Output:
+	// [{[{42 42}]} {[{41 43}]} {[{42 42}]}]
+}
+
+func ExampleDomains_assign() {
+	d1 := model.Repeat(3, model.Singleton(42))
+	d2 := d1.Assign(0, 10)
+	fmt.Println(d2)
+	// Output:
+	// [{[{10 10}]} {[{42 42}]} {[{42 42}]}]
+}
+
+func ExampleDomains_atLeast() {
+	d1 := model.Repeat(2, model.NewDomain(model.NewRange(1, 100)))
+	d2 := d1.AtLeast(1, 50)
+	fmt.Println(d2)
+	// Output:
+	// [{[{1 100}]} {[{50 100}]}]
+}
+
+func ExampleDomains_atMost() {
+	d1 := model.Repeat(2, model.NewDomain(model.NewRange(1, 100)))
+	d2 := d1.AtMost(1, 50)
+	fmt.Println(d2)
+	// Output:
+	// [{[{1 100}]} {[{1 50}]}]
+}
+
+func ExampleDomains_cmp() {
+	d1 := model.Repeat(2, model.Singleton(42))
+	d2 := model.Repeat(3, model.Singleton(43))
+	fmt.Println(d1.Cmp(d2))
+	// Output:
+	// -1
+}
+
+func ExampleDomains_domain() {
+	d := model.NewDomains(model.NewDomain(), model.Singleton(42))
+	fmt.Println(d.Domain(0))
+	fmt.Println(d.Domain(1))
+	// Output:
+	// {[]}
+	// {[{42 42}]}
+}
+
+func ExampleDomains_domains() {
```

Comment:
`Domains` is not a method on the `Domains` interface, only `Domain` is a method. I think you can delete this method or rename it to `ExampleDomains` so that it is a general example on the `Domains` interface.

Reply:
Removed